src/lu.jl

```diff
@@ -141,3 +141,9 @@ end
 
 \(F::LU, v::AbstractVector) = F.U \ (F.L \ v[F.p])
 \(F::LU, B::AbstractMatrix) = F.U \ (F.L \ B[F.p,:])
+
+function /(B::AbstractMatrix, F::LU)
+    pivot = similar(F.p)
```

Comment:
Will this allocation be elided?

Reply:
By elided you mean omitted? (Sorry, I'm not familiar with the technical terms).
If that's the case, than I think no. It is needed for the next step to be accessed at the indices F.p.
The problem is that the pivot indices cannot be used the same was as in \\. I wrote down the whole operation with matrices and the difference is that in case of / the P-matrix is multiplied from the right, while it is normally from the left (as is case of \\). So the p-indices (F.p) have to be ordered differently. There might exist a smarter way to do this but I don't know it.

Reply:
Yes, in this context, "elided" means "optimized away".

This operation with the indices looks like `Base.invperm()`. I'd suggest adding the following utility function:

```julia
Base.@propagate_inbounds function Base.invperm(p::StaticVector)
     ip = similar(p)
     ip[p] = 1:length(p)
     similar_type(p)(ip)
end
```

Checking with `@code_native` indicates that the `MVector` which is created here by `similar` is optimized away :-D And so is the range object, which is very cool and kind of surprising.

Then inside your function, use `pivot = @inbounds invperm(F.p)` (we can use `@inbounds` because we know that `F.p` is a permutation).

Reply:
Thanks, I was not aware of this function.
I added your suggestion to util. The performance increases a bit by using @inbounds.
However, in Base invperm also check if p is a valid pivot vector (by checking if every value is unique). This might be worth implementing though it probably adds computation time.

Reply:
I think we just have to assume the data in the `LU` is correct (that the pivot vector is a valid permutation).

Reply:
@c42f I'm not sure when to "trust" the optimizer - my understanding is that once the function is inlined into a more complex one, these optimizations may or may not occur (I'd love to know what is *guaranteed*...).